src/collect-all/Utilities.cs

```diff
@@ -0,0 +1,467 @@
+   
+   public GeneralException(string message) : base(message) { }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+public class Utilities
+{
+   /////////////////////////////////////////////////////////////////////////////
+   // Constructor(s)
+   /////////////////////////////////////////////////////////////////////////////
+   private Utilities(string tempPath = null, bool crossGen = false, bool loggingVerbose = false, string arch = null, string buildType = null, string coreclrRepoLocation = null) 
+   {  
+      // Find out what Operating system we are on. This is mostly just to
+      // make sure the OS is supported for SPMI Collection
+      string os = this.GetOs();;
+      
+      this.m_tempPath = tempPath;
+      this.m_crossGen = crossGen;
+      this.m_loggingVerbose = loggingVerbose;
+      
+      if (string.IsNullOrEmpty(arch) && string.IsNullOrEmpty(buildType) && string.IsNullOrEmpty(coreclrRepoLocation))
+      {
+         ParseSettingFile();
+
+         buildType = this.m_buildType;
+         coreclrRepoLocation = this.m_coreclrRepoLocation;
+         arch = this.m_arch;
+      }
+     
+      coreclrRepoLocation = this.m_coreclrRepoLocation;
+
+      // Set Default Values.
+      //
+      // jit: ryujit
+      // arch: x64
+      // buildType: Debug
+
+      if (string.IsNullOrEmpty(buildType))
+      {
+        buildType = "Debug";
+      }
+
+      if (string.IsNullOrEmpty(arch)) 
+      {
+         arch = RuntimeInformation.OSArchitecture.ToString();
+         
+         if (arch.ToLower() == "x64")
+         {
+            arch = "x64";
+         }
+         else
+         {
+            throw new GeneralException("Unsupported arch.");
+         }
+      }
+      
+      if (string.IsNullOrEmpty(coreclrRepoLocation))
+      {
+        throw new GeneralException("Error coreclrRepoLocation is empty.");
+      }
+      else
+      {
+         this.m_os = os;
+         this.m_arch = arch;
+         this.m_buildType = buildType;
+      }
+            
+      this.m_binLocation = Path.Combine(coreclrRepoLocation, "bin");
+      this.m_productLocation = Path.Combine(this.m_binLocation, "Product", this.m_os + "." + this.m_arch + "." + this.m_buildType);
+      this.m_testsLocation = Path.Combine(coreclrRepoLocation, "tests");
+      this.m_testRoot = Path.Combine(coreclrRepoLocation, "bin", "tests", this.m_os + "." + this.m_arch + "." + this.m_buildType);
+      
+      string coreRootDirName = this.m_os == "Windows_NT" ? "Core_Root" : "coreoverlay";
+      this.m_coreRoot = Path.Combine(this.m_binLocation, "tests", this.m_os + "." + this.m_arch + "." + this.m_buildType, "Tests", coreRootDirName);
+
+      if (!Directory.Exists(this.m_testRoot))
+      {
+         if (this.m_os == "Windows_NT")
+         {
+           Console.WriteLine("Error, Tests were not built. Please build CoreCLR");
```

Comment:
As written, on Windows, we will still call DownloadTests(), which runs wget, but we can't assume wget can be found on Windows machines, so this should probably be an exception